.github/workflows/npm.yml

```diff
@@ -0,0 +1,33 @@
+name: "Build and Publish NPM package"
```

Comment:
Should these be part of a **single** post-release public Action that publishes artefacts in order (easiest to revert first, i.e. push containers to MCR last) instead of parallel jobs? Otherwise, the npm package could be successfully published but building the docker containers could still fail (although mitigated by #4447).

Reply:
I discussed this with @eddyashton yesterday, I'd started doing it like that at first, but:

1. It's never going to be atomic. We can decide what goes before or after, but we cannot prevent half a release going out, and if it happens there is no easy answer. Neither MCR, NPM or PyPi can be rolled back either.
2. Splitting allows separate retries, without having to write stuff around the previous steps to make them idempotent.

Reply:
To be clear, I could still easily be convinced, there's just this mild retry-based reason to split that I can see.

Reply:
Good points. I think we can remove PyPi packages manually via the web UI (not great though) and I imagine it's also possible for NPM but having individual retries is much nicer I agree.

Reply:
There are limited circumstances where unpublishing from NPM is possible: https://docs.npmjs.com/unpublishing-packages-from-the-registry, but we cannot rely on them always being true I think.